app_dart/lib/src/request_handlers/append_log.dart

```diff
@@ -26,6 +26,7 @@ class AppendLog extends ApiRequestHandler<Body> {
     AuthenticationProvider authenticationProvider, {
     StackdriverLoggerService stackdriverLogger,
     @visibleForTesting Uint8List requestBodyValue,
+    @visibleForTesting this.key,
```

Comment:
@dnfield may have a better idea how to deal with the key here.

Reply:
I'm confused about where this comes from at runtime - it seems like it's not getting initialized to anything outside of tests.

Reply:
Yeah that's what I was trying to do, only have this accessible in tests. I added the tests for this handler in https://github.com/flutter/cocoon/pull/564 and modified the original handler to make it more testable. However, that apparently caused a few Datastore issues :)

Reply:
Making it only settable by tests is fine, but it looks like it never gets set at all in the implementation, which presumably is a problem.

Reply:
Debugging this showed that `LogChunk` should have its parent set to a Checklist key. Instead it was being set to a Task key.

I set `LogChunk` to have a parameter for setting it's parent key so it can work in the tests. I'm debugging how it was getting it's parent key set in the first place.